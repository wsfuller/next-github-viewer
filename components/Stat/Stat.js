import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Typography from '../Typography';

const Stat = ({ title, figure }) => (
  <Fragment>
    <Typography variant="h1" data-testid="statTitle">
      {title}
    </Typography>
    <Typography variant="body1" data-testid="statFigure">
      {figure}
    </Typography>
  </Fragment>
);

Stat.propTypes = {
  title: PropTypes.string.isRequired,
  figure: PropTypes.node.isRequired
};

export default Stat;
