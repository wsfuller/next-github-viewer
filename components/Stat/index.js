import React, { Fragment } from 'react';
import styled from 'styled-components';

const StyledH3 = styled.h3`
  font-size: 36px;
`;

const Paragraph = styled.p`
  color: ${({ theme }) => {
    return 'blue';
  }};
`;

const Stat = ({ title, figure }) => (
  <Fragment>
    <StyledH3>{title}</StyledH3>
    <Paragraph>{figure}</Paragraph>
  </Fragment>
);

export default Stat;
