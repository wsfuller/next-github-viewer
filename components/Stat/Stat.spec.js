import { shallow, mount, render } from 'enzyme';
import Stat from '.';

// const wrapper = shallow(<Stat />);

test('something interesting', () => {
  expect(true).toEqual(false);
});
