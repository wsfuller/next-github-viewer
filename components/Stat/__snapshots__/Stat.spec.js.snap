// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stat /> renders correctly 1`] = `
Array [
  <h1
    className="Typographystyles__StyledH1-sc-1n6ui1k-0 bvGdAG"
    data-testid="statTitle"
    transform={null}
  >
    Stat Title
  </h1>,
  <p
    className="Typographystyles__StyledBody-sc-1n6ui1k-5 cdoZfY"
    data-testid="statFigure"
    transform={null}
  >
    55
  </p>,
]
`;
